### DISCUSSION TITLE
Feature Request: +Intersect Drawing (Video Included)

### DISCUSSION THREAD

**Discussion by tingham** [keiwando/toonsquid#113] [Ideas]
I think that this would be a supreme addition to your application. It would save me a ridiculous amount of time and handwork as a third type of frame creation.

If I'm timing between two frames and I know that my result is going to be equidistant there are two "edge" cases (sorry, couldn't resist the pun.)

The first is in the situation where I'm moving an object in place or doing a half-move. In these situations, as shown visually in the video below there ends up being a lot of overlapping material between the two frames. If you look at the onions of the boundary frames you can really see what I'm talking about. I've outlined it at https://youtu.be/izs1lRSumgQ?t=58

*The second case is a whole lot more complicated where you essentially raycast through the vertices of the paths used to create the material on the frame to determine halfway points and I'm **NOT** trying to get you to go down that rabbit hole with this request.*

I would propose a new menu item "+Intersect Drawing" between "+Add Drawing" and "+Split Drawing" that only appears conditionally if:

* Each frame has a ratio of (filled pixels) to (empty pixels) - this does not need to take alpha into account. A Fill-Rate of 1.0 would mean that the frame is entirely covered with color data in every pixel. Fill-Rate zero means that there is nothing in the bitmap for a frame.
* If the multiplication of the two boundary frame Fill-Rates is greater than zero, show the menu item.

When selected the menu item would:

1. Evaluate the delta between the boundary frames
2. Create a new drawing at the currently selected frame
3. Produce pixel brush output with the current brush settings for color and opacity where...
4. Only the overlapping areas of the onion skin boundary frames would show a color mix

From this point the user can utilize the onion skins as normal to fill out the new tween shape, using the intersection art as a guide. Once the intermediate frame is created, timing can be controlled as normal.

It is incredibly likely that one this sort of tool/evaluation pattern existed - we (the community) would thinking a bunch of other ways to use similar tooling. Seems pretty exciting.

Full video:

https://www.youtube.com/watch?v=izs1lRSumgQ

(Please ignore my crummy drawing, I did this specifically to show the feature idea)

**Comment by keiwando** [comment]
Thank you very much for the suggestion and also for taking the time to record that video!

If I understood the request correctly, you would like an option that can create an in-between and that immediately fills all pixels that are not fully transparent in both the previous and next drawing in the same layer, right? 

Optimally the pixels would get filled by brush strokes generated with the selected brush, however that of course also adds a whole lot of complexity to the feature. (Maybe I also misunderstood you here and you only meant that the selected color and brush strength should be used to fill those pixels and not the actual brush itself)

The feature definitely sounds interesting, but since it is currently very difficult for me to determine how to prioritize it, I would love to hear more input from a lot of other artists on whether this would actually benefit their workflow as well.

**Reply by tingham** [reply]
Fill the pixels as if the currently selected brush and color were input by the pencil at 100% pressure.

**Comment by DinoAwakening** [comment]
It took me a while to comprehend what was he asking for (Since it’s a 7 minutes-length video). But yeah, it would save us tons and tons of time, just being able to make an in between of our bitmaps layers. I was about to ask for that last time, but at first I thought it was to much to ask for. Nevertheless, now I saw your opinion about it, I’m glad to support (Now this guy’s) Idea. :D

**Reply by tingham** [reply]
Sorry about the video length. I'm still not used to editing videos on the ipad and had enough trouble squashing it down from the original 30 minutes or whatever.
